client/types.go

```diff
@@ -91,6 +108,27 @@ type Hierarchy struct {
 	} `xml:"metric"`
 }
 
+type UsageReports map[string]UsageReport
+
+// UsageReport - captures the XML response for rate limiting details
+type UsageReport struct {
+	Period       LimitPeriod
+	PeriodStart  int64
+	PeriodEnd    int64
+	MaxValue     int
+	CurrentValue int
+}
+
+// UsageReportXML - captures the XML response for rate limiting details
+type UsageReportXML struct {
+	Metric       string      `xml:"metric,attr"`
+	Period       LimitPeriod `xml:"period,attr"`
+	PeriodStart  string      `xml:"period_start"`
+	PeriodEnd    string      `xml:"period_end"`
+	MaxValue     int         `xml:"max_value"`
+	CurrentValue int         `xml:"current_value"`
+}
```

Comment:
is there no way to avoid storing these and have them converted directly into the `UsageReport` struct above?

Reply:
Through some custom unmarshal function perhaps, but honestly, for xml its just easier to convert this. I'll look to see later if I can unmarshal the inner field only

Reply:
Plus I wanted to turn this into a map also since it is easier to pull out the data we are interested in